File: case study subjects/StringParser.cs
bool cascade = false , inValue = false;
String currentValue = "";
List <String > values = new List <String >();
for (int i = 0; i < s.Length; i++) {
char c = s[i];

if (c =='\\' && !cascade) { 

cascade = !cascade; }

else if (c == ''' && !cascade) { 

inValue = !inValue; }

else if (c == ',')
 { 
 
 values.Add(currentValue); currentValue = "";}
 
 else { 
 
 currentValue += c; }
 cascade = false;
 
 }
 }
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 

File: case study subjects/String-parser.java
int  Parser(int A[5]) {       
	
		    bool cascade;  bool inValue ;
			int i;
	        cascade = false;
			inValue = false;
		    i = 0;
			for (i = 0; i < 5; i++) {

				if ((A[i] == 92) && !cascade )
					
					cascade = !cascade;
				
				else if ((A[i] == 34) && !cascade)

				
				else if (A[i] == 44) 

				else 
				
				cascade = false;
	
			}
}